File: Java/Program/Film.java
public class Film {
  private int id;
  private String title;
  private String genre;
  private double rating;
  private int release;
  private int duration;

  public Film(int id, String title, String genre, double rating, int release, int duration) {
    this.id = id;
    this.title = title;
    this.genre = genre;
    this.rating = rating;
    this.release = release;
    this.duration = duration;
  }

  // mengembalikan id film (untuk update dan hapus data film)
  public int getFilmId() {
    return id;
  }

  // menampilkan detail data film
  public void showFilm() {
    System.out.println("ID: " + id + " | Judul: " + title + " | Genre: " + genre + " | Rating: " + rating + " | Rilis: " + release + " | Durasi: " + duration + " mnt");
  }

  // mengubah data film
  public void updateFilm(String newTitle, String newGenre, double newRating, int newRelease, int newDuration) {
    this.title = newTitle;
    this.genre = newGenre;
    this.rating = newRating;
    this.release = newRelease;
    this.duration = newDuration;
  }
}

File: PHP/Film.php
<?php
// class film dengan encapsulation
class Film {
  private $id;
  private $title;
  private $genre;
  private $rating;
  private $release;
  private $duration;
  private $image;

  public function __construct($id, $title, $genre, $rating, $release, $duration, $image) {
    $this->id = $id;
    $this->title = $title;
    $this->genre = $genre;
    $this->rating = $rating;
    $this->release = $release;
    $this->duration = $duration;
    $this->image = $image;
  }

  // mengembalikan id film (untuk update dan hapus data film)
  public function getFilmId() {
    return $this->id;
  }

  public function getTitle() {
    return $this->title;
  }

  public function getGenre() {
    return $this->genre;
  }

  public function getRating() {
    return $this->rating;
  }

  public function getRelease() {
    return $this->release;
  }

  public function getDuration() {
    return $this->duration;
  }

  public function getImage() {
    return $this->image;
  }

  // mengubah data film
  public function updateFilm($newTitle, $newGenre, $newRating, $newRelease, $newDuration, $newImage) {
    $this->title = $newTitle;
    $this->genre = $newGenre;
    $this->rating = $newRating;
    $this->release = $newRelease;
    $this->duration = $newDuration;
    if ($newImage != "") {
      $this->image = $newImage;
    }
  }
}
?>